Unit tests on buildbots

maaktweluit opened the pull request:
Resolves #1400 

~Depends on #1743~
~Depends on #1764~

- Moved all unit tests to buildbot
- Added inno installer to the `buildpackage_windows` job
- Moved linter tests to buildbot
- Lowered github status count to 2, `test` and `build`
- Increased workers to 2 per OS ( + 2 for linting )

Open Tasks:
- [x] Deploy production servers
  - [x] Setup AWS
  - [x] Tweak config
  - [x] Install AWS
  - [x] Testing
  - [x] Switch public DNS record
- [x] Finish nightly build script
- [x] Fix rebuild protection slow vs fast

Possible follow up improvements:
- [x] Kill hyperg instead of using old process for tests, fail if not successful #1900 
- [x] Facilitate updates of dependencies ( for unit test and installer )
- [x] Make master redundant
- [x] Split `golem-dev` role from `buildbot-worker` on windows and mac, like it is on linux
- [x] Store lint results of develop and download to compare. Now it runs 2x lint per run, develop and branch #1794

maaktweluit commented:
Made issues for:
- Nightly uploads
- Redundant master

Will try to fix the rebuild-slow and rebase to get this finished PR merged.

maaktweluit commented:
Updated the code to the latest of develop and made all follow up issues.

Now this PR is ready to be reviewed and merged :)

ktomala commented on scripts/ci/roles/buildbot-worker-linux/tasks/main.yml:
```diff
@@ -2,168 +2,78 @@
+    executable: pip3
     name: buildbot-worker
-  become_user: buildbot-worker
+  become_user: "{{worker_user}}"
```
According to ANSIBLE0017 you should probably add "become: yes" also

maaktweluit replied:
Never knew there was an Ansible set of `PEP001` rules as well, awesome!

Since it is already running and in production i do not feel safe adding this and running it again. Durning installation there where some problems with `become: yes`, though mostly on mac and windows.

etam replied:
http://docs.ansible.com/ansible/latest/become.html
> become_user - [...] Does NOT imply become: yes

maaktweluit replied:
When trying to fix this i found there is a bug in `ANSIBLE0017`: https://github.com/willthames/ansible-lint/issues/215

in `site-linux-worker.yml` become is set for the full role.

ktomala replied:
Should we ignore ANSIBLE0017 for now?

maaktweluit replied:
Yes, think it is best.

We should find a solve for both mac and linux separately before enabling this check.

ktomala commented on scripts/ci/roles/buildbot-worker-linux/tasks/main.yml:
```diff
@@ -2,168 +2,78 @@
-    creates=/home/buildbot-worker/worker/
-  become_user: buildbot-worker
+    creates="{{_worker_service_dir}}"
+  become_user: "{{worker_user}}"
```
According to ANSIBLE0017 you should probably add "become: yes" also

maaktweluit replied:
replied above

ktomala commented on scripts/ci/roles/buildbot-worker-linux/tasks/main.yml:
```diff
@@ -2,168 +2,78 @@
+    name: user.email
+    scope: global
+    value: "{{worker_name}}@buildbot.local"
+  become_user: "{{worker_user}}"
```
According to ANSIBLE0017 you should probably add "become: yes" also

maaktweluit replied:
replied above

ktomala commented on scripts/ci/roles/buildbot-worker-linux/tasks/main.yml:
```diff
@@ -2,168 +2,78 @@
+    name: user.name
+    scope: global
+    value: "{{worker_name}}"
+  become_user: "{{worker_user}}"
```
According to ANSIBLE0017 you should probably add "become: yes" also

ktomala commented on scripts/ci/roles/buildbot-worker-mac/tasks/main.yml:
```diff
@@ -0,0 +1,119 @@
+
+#install geth
+- name: add geth repo
+  become_user: buildbot-worker
```
According to ANSIBLE0017 you should probably add "become: yes" also

ktomala commented on scripts/ci/roles/buildbot-worker-mac/tasks/main.yml:
```diff
@@ -0,0 +1,119 @@
+    name: ethereum/ethereum
+
+- name: install geth and dependencies
+  become_user: buildbot-worker
```
According to ANSIBLE0017 you should probably add "become: yes" also

maaktweluit replied:
on mac this does not work as intended, maybe we have to re-configure become-method.

Propose to investigate this when splitting up the roles.

ktomala commented on scripts/ci/roles/buildbot-worker-mac/tasks/main.yml:
```diff
@@ -0,0 +1,119 @@
+
+# install hyperg
+- name: install tar
+  become_user: buildbot-worker
```
According to ANSIBLE0017 you should probably add "become: yes" also

ktomala commented on scripts/ci/roles/buildbot-worker-mac/tasks/main.yml:
```diff
@@ -0,0 +1,119 @@
+
+# install pythons
+- name: install python and deps
+  become_user: buildbot-worker
```
According to ANSIBLE0017 you should probably add "become: yes" also

ktomala commented on scripts/ci/site-worker-control.yml:
```diff
@@ -0,0 +1,29 @@
+  become: yes
+  gather_facts: False
+  vars:
+    - master_host: {{hostvars[groups['master'][0]]['ec2_private_ip_address']}}
```
Please wrap template variable in double quotes, i.e.:
``` - master_host: "{{hostvars[groups['master'][0]]['ec2_private_ip_address']}}"```

maaktweluit replied:
Tried to keep everything without "" if it would still run, is that not a good practice?

etam replied:
http://docs.ansible.com/ansible/latest/playbooks_variables.html#using-variables-about-jinja2

ktomala commented on scripts/ci/site-worker-mac.yml:
```diff
@@ -0,0 +1,8 @@
+  remote_user: admin
+  roles:
+    - role: buildbot-worker-mac
+      worker_master_hostname: {{hostvars[groups['master'][0]]['ec2_public_ip_address']}}
```
Please wrap template variable in double quotes, i.e.:
``` worker_master_hostname: {{hostvars[groups['master'][0]]['ec2_public_ip_address']}}```

ktomala commented on scripts/ci/site-worker-windows.yml:
```diff
@@ -0,0 +1,9 @@
+  become_method: runas
+  roles:
+    - role: buildbot-worker-windows
+      worker_master_hostname: {{hostvars[groups['master'][0]]['ec2_private_ip_address']}}
```
Please wrap template variable in double quotes, i.e.:
```  worker_master_hostname: "{{hostvars[groups['master'][0]]['ec2_private_ip_address']}}" ```

maaktweluit commented:
Thank you all for the great comments!

Replied to all and updated the PR.

Also added 1 small fix after the "pull request webhooks" got enabled and broke ( SHA c66ee6a )
